joomla/joomla-cms
# [4.0] Housekeeping

Pull Request for Issue # .

### Summary of Changes
The core.js is supposed to be the JS API of Joomla but we keep sneaking methods there with the expectation that these will be useful to somebody...
Since js is becoming more modular inflating some file is against the common practice. So there you have it, methods that are specific to some other functionality are moved to the related files.
Moved: 
- `Joomla.iframeButtonClick` The method applies only to iframes in modals, so the method is moved to the modal initiator file (legacy/bootstrap-init.js)

- `Joomla.localStorageEnabled` The method is used **only** in the backend menu so it's moved there

- `Joomla.resetFilters` The method works **only** on the search tools (filters) so it's also moved there (searchtools.js)

- Consistenly use `Joomla.Text` instead of the `Joomla.JText`. (the later is still working but deprecated)

### Testing Instructions
Check in the backend that:
- The menu still works
- Try in the Menus the Site/Administrator dropdown
- For the modal close you need to disable the redirect plugin, then go to the System->redirects and on the message that says that the plugin needs to be enabled click on the link. A modal will open change the settings then save and close. Then refresh the page. The plugin should be enabled

### Expected result



### Actual result

### TODO

- `Joomla.renderMessages` needs to be cleaned as J4 has only one messaging system so no point supporting also the Bootstrap.

- `Joomla.Event` that it was introduced in J4 some time ago needs to be **completely removed**
There is zero benefit having a shortcut for the native methods:
![screenshot 2019-01-20 at 15 04 37](https://user-images.githubusercontent.com/3889375/51440193-bc4f6b80-1cc4-11e9-9c32-235c1c06b7ed.png)
@Fedik do you agree on this one, as you're the one that introduced it?

### Documentation Changes Required
None
